src/watchman_watcher.js

```diff
@@ -69,16 +69,18 @@ WatchmanWatcher.prototype.init = function() {
     return self.watchProjectInfo ? self.watchProjectInfo.root : self.root;
   }
 
-  function onVersion(error, resp) {
+  function onCapability(error, resp) {
     if (handleError(self, error)) {
+      // The Watchman watcher is unusable on this system, we cannot continue
+      process.exit(1);
```

Comment:
I was leaving the decision making to the application because sometimes it's not straightforward. @stefanpenner you had code that did this? Want to send a PR?